covid19-etl/etl/cityofchicago.py

```diff
@@ -0,0 +1,410 @@
+            )
+
+            for row in reader:
+                self.parse_row(row)
+
+    def parse_row(self, row):
+
+        """
+        according to row mapping in the dataset, in each row
+        column 0 would be lab report date
+        column 1,2 and 3 would be total number of cases, deaths and hospitalization per day which would be used in summary_clinical
+        column 3 to rest would be for summary_group_demographics for Age group, race , gender and ethincity
+        Here we are ignoring records which doesn't have any lab report dates
+        """
+        if row[0] != "":
+            row_dict = {}
+            file_latest_date = self.get_unified_date_format(row[0])
+            row_dict[self.headers[0]] = file_latest_date
+            summary_clinical_submitter_id = self.add_summary_clinical(
+                self.convert_str_to_int(row[1]),
+                self.convert_str_to_int(row[2]),
+                self.convert_str_to_int(row[3]),
+                datetime.strptime(file_latest_date, "%Y-%m-%d"),
+            )
+            for i in range(3, len(self.headers)):
+                row_dict[self.headers[i]] = self.convert_str_to_int(row[i])
+                self.add_to_summary_group_demographics(
+                    self.headers[i],
+                    row_dict[self.headers[i]],
+                    summary_clinical_submitter_id,
+                )
+
+    def files_to_submissions(self):
+        # ETL code that reads from the data source
+        # and generates the data to submit
+        start = time.time()
+        latest_submitted_date = self.metadata_helper.get_latest_submitted_date_idph()
+
+        # The following condition is for the first entry in dataset, which is from date `03-01-2020`
+        if latest_submitted_date == None:
+            latest_submitted_date = datetime.strptime("2020-03-01", "%Y-%m-%d")
+
+        today = datetime.today()
+        today = today.replace(hour=0, minute=0, second=0, microsecond=0)
+
+        if latest_submitted_date == today:
+            print("Nothing to submit: today and latest submitted date are the same.")
+            return
+
+        print(
+            f"Latest submitted date: {latest_submitted_date}. Getting data until date: {today}"
+        )
+
+        self.summary_location_submitter_id = format_submitter_id(
+            "summary_location",
+            {"country": self.country, "state": self.state, "city": self.city},
+        )
+
+        self.get_summary_location()
+        self.parse_cityofchicago_file(
+            latest_submitted_date.strftime("%Y-%m-%dT%H:%M:%S"),
+            today.strftime("%Y-%m-%dT%H:%M:%S"),
+        )
+        print("Done in {} secs".format(int(time.time() - start)))
+
+    def submit_metadata(self):
+        # Submits the data in `self.summary_locations`, `self.summary_clinicals` and `self.summary_group_demographic` to Sheepdog.
+        print("Submitting data...")
+        if len(self.summary_locations) > 0:
```

Comment:
this `if` is not necessary, `for sl in self.summary_locations.values()` will just do nothing if the dict is empty